CMakeLists.txt

```diff
@@ -0,0 +1,124 @@
+#*****************************************************************************
+# DESCRIPTION: Script for build tool cmake on both unix and windows
+#
+# This file is part of Verilator.
+#
+# Code available from: https://verilator.org
+#
+#*****************************************************************************
+#
+# Copyright 2003-2022 by Wilson Snyder. This program is free software; you
+# can redistribute it and/or modify it under the terms of either the GNU
+# Lesser General Public License Version 3 or the Perl Artistic License
+# Version 2.0.
+# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+#
+#****************************************************************************/
+
+cmake_minimum_required(VERSION 3.15)
+cmake_policy(SET CMP0091 NEW) # Use MSVC_RUNTIME_LIBRARY to select the runtime
+project(Verilator
+    VERSION 5.003
+    HOMEPAGE_URL https://verilator.org
+    LANGUAGES CXX
+)
+
+option(CMAKE_UNITY_BUILD
+    "Builds concatenated source files, potentially sppeding up the build.")
+option(DEBUG_AND_RELEASE_AND_COVERAGE
+    "Builds both the debug and release binaries, overriding CMAKE_BUILD_TYPE. Not supported under MSBuild.")
+
+set(CMAKE_INSTALL_DATADIR ${CMAKE_INSTALL_PREFIX})
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(CheckStructHasMember)
+include(ExternalProject)
+
+if (MSVC)
```

Comment:
Add somewhere for Linux builds

message(FATAL_ERROR "Use gmake instead of cmake for Linux builds.")

Reply:
cmake on linux is supported

Reply:
I appreciate the code you added here works on Linux.  Maybe a better way to describe it is I would like a warning this is "experimental" on Linux.

When this code has CI testing, has been proven through several releases,  months of user usage, and we've seen that you and others here can support the bugs and questions that people create, then it seems a good alternative to the working Linux make scripts and that warning can be removed.   For MSWIN it is a lower bar as there we do not have a long-proven working alternative in make.

Reply:
added warning message